README.md

```diff
@@ -1,32 +1,506 @@
+            {
+                path: "/about",
+                element: <About />
+            },
+            {
+                path: "/login",
+                element: <Login />
+            },
+            {
+                path: "/profile/:id",
+                element: <UserProfile />
+            }
+        ]
+    }
+];
+
+export default routes;
+```
+
+By entering our different route objects as an array associated with our `App`
+route's `children` key, we've set them up to render _inside_ of our `App`
+component. That means that if we navigate to any of these _nested routes_ - such
+as `/login`, for example - our `App` component will render with our `Login`
+component as a child component.
+
+Note that it's ok for our `Home` component to have the same path as our parent
+`App` component. All child route paths _must_ start with their parent's route
+path, but they can also directly match their parent's route path. However, you
+can only have one child route whose path exactly matches its parent's path - you
+can't have multiple child routes whose paths exactly match the parent's path.
+
+Now that all of our routes are children of `App`, we can just include our
+`errorElement` on `App` - any errors that occur in one of our nested routes will
+"bubble up" to the parent route, which will render our ErrorPage. Much DRYer!
+
+You can still render unique errorElements for each route, if you want to create
+different error handling pages for different routes.
+
+Since `App` now renders no matter what URL we visit, we can also just include
+our `NavBar` component directly within our `App`, rather than dropping it into
+every page-level component:
+
+```jsx
+// App.js
+import NavBar from "./components/NavBar";
+
+function App(){
+    return(
+        <>
+            <header>
+                <NavBar />
+            </header>
+        </>
+    );
+};
+```
+
+Much easier! And, if we create a new page for our website, we don't have to
+remember to include the `NavBar` component within that new page.
+
+Remember to remove the `NavBar` portion of your JSX from the `Home` component
+after adding this code to `App`.
+
+## Using react-router-dom's Outlet Component
+
+If you've opened the code up in your browser, you might have noticed that our
+app still isn't working correctly - visiting the child routes doesn't actually
+render the pages we want.
+
+That's because there is still one tool we need to implement from
+`react-router-dom` in order to get our nested routes up and running - the
+`Outlet` component.
+
+An `Outlet` component is included within a component that has nested routes. It
+basically serves as a signal to that parent component that it will render
+various different components as its children, depending on what route a user
+visits. The `Outlet` component works in conjuction with the `router` we set up
+using `createBrowserRouter` and `RouterProvider` to determine which component
+should be rendered based on the current route.
```

Comment:
Hmm, good catch. I don't know exactly how they work under the hood, but my hypothesis is that the `router` would register the route, find the appropriate parent / child components for that route, then render them. It may check to see if any parent routes match the start of the URL, then check if those parent routes are rendering Outlets. But, I'm not 100% sure.

If we don't think this sentence adds clarity, we can just remove it. I'm not sure how else we would phrase this.

Reply:
How about "The `Outlet` component works in conjunction with the `router` to determine...."?

Reply:
So just remove the "we set up
using `createBrowserRouter` and `RouterProvider`" portion?